### PR TITLE
chore(actor) Move Rule reads from owner_id to team/user columns

### PR DESCRIPTION
**Pull request by markstory**
Move read operations off of `owner_id` and use `team_id` and `user_id` instead.

Refs HC-1171

### CODE REVIEW

File: src/sentry/deletions/defaults/team.py [hunk 1]
```diff
@@ -21,7 +21,7 @@ def delete_instance(self, instance):
         from sentry.models.rule import Rule
         from sentry.monitors.models import Monitor
 
-        AlertRule.objects.filter(owner_id=instance.actor_id).update(owner=None)
-        Rule.objects.filter(owner_id=instance.actor_id).update(owner=None)
-        Monitor.objects.filter(owner_team_id=instance.actor_id).update(owner_team_id=None)
+        AlertRule.objects.filter(team_id=instance.id).update(owner=None, team_id=None)
+        Rule.objects.filter(owner_team_id=instance.id).update(owner=None, owner_team_id=None)
+        Monitor.objects.filter(owner_team_id=instance.id).update(owner_team_id=None)
```
Comment (markstory) [inline src/sentry/deletions/defaults/team.py:26]: The previous condition did `owner_team_id=instance.actor_id` which I don't think is correct and I've adjusted in these changes.

File: src/sentry/incidents/endpoints/organization_incident_index.py [hunk 2]
```diff
@@ -97,11 +97,9 @@ def get(self, request: Request, organization) -> Response:
             except InvalidParams as err:
                 return Response(str(err), status=status.HTTP_400_BAD_REQUEST)
 
-            team_filter_query = Q(
-                alert_rule__owner_id__in=teams_query.values_list("actor_id", flat=True)
-            )
+            team_filter_query = Q(alert_rule__team_id__in=teams_query.values_list("id", flat=True))
             if unassigned:
-                team_filter_query = team_filter_query | Q(alert_rule__owner_id=None)
+                team_filter_query = team_filter_query | Q(alert_rule__team_id=None)
```
Comment (wedamija) [inline src/sentry/incidents/endpoints/organization_incident_index.py:102]: I wonder if this worked before... we'd usually use `alert_rule__owner_id__isnull=True`

**Review by wedamija — APPROVED** [review]

File: src/sentry/deletions/defaults/team.py (hunk 1 above)
Comment (evanpurkhiser) [reply src/sentry/deletions/defaults/team.py:26]: Thanks for cleaning this up

File: src/sentry/api/serializers/models/rule.py [hunk 3]
```diff
@@ -134,22 +130,11 @@ def get_attrs(self, item_list, user, **kwargs):
...
-
         for rule in rules.values():
-            if rule.owner_id:
-                type = actor_type_to_string(rule.owner.type)
-                if rule.owner_id in resolved_actors[type]:
-                    result[rule]["owner"] = f"{type}:{resolved_actors[type][rule.owner_id]}"
+            if rule.owner_team_id or rule.owner_user_id:
+                actor = ActorTuple.from_id(user_id=rule.owner_user_id, team_id=rule.owner_team_id)
+                if actor:
+                    result[rule]["owner"] = actor.identifier
```
Comment (evanpurkhiser) [inline src/sentry/api/serializers/models/rule.py:137]: In crons we serialized this using the actor serializer. Should we do that here too?

Comment (evanpurkhiser) [reply src/sentry/api/serializers/models/rule.py:137]: https://github.com/getsentry/sentry/blob/c601acee1b344a608ea4b491d7cdc5002cc7db5b/src/sentry/monitors/serializers.py#L184-L191

https://github.com/getsentry/sentry/blob/c601acee1b344a608ea4b491d7cdc5002cc7db5b/src/sentry/monitors/serializers.py#L220

File: src/sentry/incidents/endpoints/organization_incident_index.py (hunk 2 above)
Comment (markstory) [reply src/sentry/incidents/endpoints/organization_incident_index.py]: I'll switch to `isnull`.

**Pull request merged**